src/transfer.jl

```diff
@@ -168,6 +76,70 @@ function transfer_blochwave_kpt(ψk_in, basis_in::PlaneWaveBasis{T}, kpt_in::Kpo
     ψk_out
 end
 
+"""
+Transfer an array `ψk_in` expanded on `kpt_in`, and produce ``ψ(r) e^{i ΔG·r}`` expanded on
+`kpt_out`. It is mostly useful for phonons.
+Beware: `ψk_out` can lose information if the shift `ΔG` is large or if the `G_vectors`
+differ between `k`-points.
+"""
+function transfer_blochwave_kpt(ψk_in, basis::PlaneWaveBasis, kpt_in, kpt_out, ΔG)
+    ψk_out = zeros(eltype(ψk_in), length(G_vectors(basis, kpt_out)), size(ψk_in, 2))
+    for (iG, G) in enumerate(G_vectors(basis, kpt_in))
+        # e^i(kpt_in + G)r = e^i(kpt_out + G')r, where
+        # kpt_out + G' = kpt_in + G = kpt_out + ΔG + G, and
+        # G' = G + ΔG
+        idx_Gp_in_kpoint = index_G_vectors(basis, kpt_out, G - ΔG)
+        if !isnothing(idx_Gp_in_kpoint)
+            ψk_out[idx_Gp_in_kpoint, :] = ψk_in[iG, :]
+        end
+    end
+    ψk_out
+end
+
+"""
+Real-space version of the previous function.
+"""
+function transfer_blochwave_kpt_real(ψk_in, basis::PlaneWaveBasis, kpt_in, kpt_out, ΔG)
+    ψk_out = zeros(eltype(ψk_in), length(kpt_out.G_vectors), size(ψk_in, 2))
+    exp_ΔGr = cis2pi.(-dot.(Ref(ΔG), r_vectors(basis)))
+    for n in 1:size(ψk_in, 2)
+        ψk_out[:, n] = fft(basis, kpt_out, exp_ΔGr .* ifft(basis, kpt_in, ψk_in[:, n]))
+    end
+    ψk_out
+end
+
+"""
+Find the equivalent index of the coordinate `kcoord` ∈ ℝ³ in a list `kcoords` ∈ [-½, ½)³.
+`ΔG` is the vector of ℤ³ such that `kcoords[index] = kcoord + ΔG`.
+"""
+function find_equivalent_kpt(kcoords::Vector{Vec3{T}}, kcoord; tol=100*eps(T)) where {T}
```

Comment:
It somehow feels we have this or a similar function somewhere in the code already. Have you checked e.g. the symmetry handling when we extract data from spglib. I think we use something along these lines there. It would be good in that case to generalise and only have the code once.

Reply:
We don't actually do this anywhere else. Closest is @LaurentVidal95 in the wannier code, but he actually takes that information from the nnkpt file produced by wannier90, so he doesn't have it directly